### ISSUE TITLE
Broken language codes

### ISSUE THREAD

**Issue by skalee** [geolexica/isotc211-glossary#16]
Some languages are broken after migration. 

- The correct ISO code is `zho`, but this repo uses `chi`.
- The correct ISO code is `nld`, but this repo uses `dut`.
- The correct ISO code is `deu`, but this repo uses `ger`.

They have been fixed in Excel recently (as in https://github.com/geolexica/isotc211.geolexica.org/issues/110).

**Comment by ronaldtse** [comment]
Maybe this is related to Glossarist? Ping @strogonoff

**Comment by ronaldtse** [comment]
@strogonoff will update this in the Glossarist Desktop.

**Comment by strogonoff** [comment]
@ronaldtse Can we change IEV to match the codes?

**Comment by strogonoff** [comment]
Unassigning myself as codes in Glossarist Desktop is a separate concern (https://github.com/glossarist/glossarist-desktop/issues/49)

**Comment by ronaldtse** [comment]
> @ronaldtse Can we change IEV to match the codes?

Yes!

**Comment by skalee** [comment]
Do you mean that you want to replace valid ISO 639-2 codes used in IEV with invalid ones used in TC211 glossary?

**Comment by ronaldtse** [comment]
I meant we should change everything to the correct codes.

**Comment by strogonoff** [comment]
> invalid ones used in TC211 glossary 

I thought ISO/TC 211 glossary uses correct language codes as of https://github.com/geolexica/isotc211.geolexica.org/issues/110 🤔

**Comment by skalee** [comment]
They have been fixed in spreadsheets recently (https://github.com/geolexica/isotc211.geolexica.org/issues/110 indeed), but these changes were somehow reverted at data migration.

**Comment by strogonoff** [comment]
Ah was that because Glossarist Desktop was still using the wrong language codes at the time of migration? Sorry if that.

Now that the app was updated to use the correct codes, let me run another migration against my fork of this repo. I expect the diff should be much smaller, since YAML is going to be dumped using the same formatting. I’ll check if that’s the case and will make a PR.

The other way to fix this would be to run a Ruby script to migrate YAML. If anyone is on to that already, please do let me know (then I won’t do language migration in the app)

**Comment by ronaldtse** [comment]
I think a global regex replace would be super easy...?

**Comment by skalee** [comment]
I can do a pull request.

**Comment by skalee** [comment]
Wow, I'm confused. It looks like this repo already uses corrected codes, both in [`tc211-termbase.meta.yaml`](https://github.com/geolexica/isotc211-glossary/blob/028327a/tc211-termbase.meta.yaml) and in [concepts](https://github.com/geolexica/isotc211-glossary/blob/1e9a29d483c4f720606535c3f13502ad9001fdfa/concepts/concept-145.yaml). Also removing my temporary fixes in TC 211 site does not break anything.

I don't know, maybe I was working on some broken branch of the glossary or something…

**Comment by strogonoff** [comment]
Oh right, indeed. @skalee Could you confirm there are no leftover wrong language codes on some concepts? This should be doable with a grep or a Ruby script, I suppose…

**Comment by skalee** [comment]
Yes, I confirm that there are no broken codes in `concepts/concept-*.yaml` nor in `tc211-termbase.meta.yaml` nor in `register.yaml` nor in `change-requests/*`. I suppose this issue should be closed now. Thank you!
